### PR TITLE
Add pretty printing back to debugger without unwanted blank lines

### PR DESCRIPTION
**Pull request by goose121**
This adds pretty-printing again without the unwanted blank lines before output introduced in the previous attempt.

### CODE REVIEW

File: src/debugger.lisp [hunk 1]
```diff
@@ -72,32 +95,36 @@ object the debugger was entered with."
...
                  for restart-name = (or (restart-name restart) "")
-                 do (format stream ";; ~2,' D: [~vA] ~A~%"
-                            i max-name-length restart-name report)))
-          (t (format stream "~&;; No available restarts.~%")))))
+                 do (format stream "~%~2,' D: [~vA] "
+                            i max-name-length restart-name)
+                    (handler-case
+                        (format stream "~@<~A~:>" restart)
+                      (error ()
+                        (format stream "#<error while reporting restart>")))))
```
Comment (phoe) [inline src/debugger.lisp:116]: Why is this duplicated with L80 in the old code? We already have the `report` variable, is it unused now?

File: t/debugger.lisp [hunk 2]
```diff
@@ -57,7 +57,10 @@
 
 (deftest debugger.condition.1
   (eqt (first (portable-condition-system::run-debugger-command
-               :condition nil *debugger.condition*))
+               :condition
+               ;; Dummy stream that supports both input and output
+               (make-two-way-stream (make-concatenated-stream) (make-broadcast-stream))
```
Comment (phoe) [inline t/debugger.lisp]: Does it mean that passing `nil` to `run-debugger-command` is now unsupported?

If yes, then this stream needs to be created internally, and changes to this test case should be reverted - we should support having `nil` as a valid argument when no streams are required from the outside, because the user does not always have a meaningful stream to pass.

**Review by phoe — CHANGES_REQUESTED** [review]

File: t/debugger.lisp (hunk 2 above)
Comment (goose121) [reply t/debugger.lisp]: Oh okay, can do; it will cause a condition to be signalled in the current state if nil is passed

File: src/debugger.lisp (hunk 1 above)
Comment (goose121) [reply src/debugger.lisp:116]: Oh whoops, l80 should be deleted

File: src/debugger.lisp [hunk 3]
```diff
@@ -10,11 +10,38 @@ and printing output to the provided stream. The condition object the debugger
...
+  executing a command.")
+
+(defmethod run-debugger-command :around (command stream condition &rest arguments)
+  "Prefixes each line of output with semicolons unless told not to."
+  (cond
+    (*skip-debugger-prefix*
+     (call-next-method))
+    (t
+     (let* ((stream (or stream (make-two-way-stream (make-concatenated-stream) (make-broadcast-stream))))
+            (stream-out stream))
```
Comment (phoe) [inline src/debugger.lisp:24]: This second variable seems a bit redundant. Why not just `stream`?

File: src/debugger.lisp [hunk 4]
```diff
@@ -10,11 +10,38 @@ and printing output to the provided stream. The condition object the debugger
 was entered with and optional command arguments are available for use within
 the command itself."))
 
+(defvar *skip-debugger-prefix* nil
+  "Whether or not to avoid printing the debugger line prefix when
+  executing a command.")
```
Comment (phoe) [inline src/debugger.lisp]: Please don't pad documentation strings with spaces, and try to break close to 80 chars per line - that's consistent with this repository's style.

**Review by phoe — CHANGES_REQUESTED** [review]

File: src/debugger.lisp (hunk 3 above)
Comment (goose121) [reply src/debugger.lisp:24]: `pprint-logical-block` would rebind the `stream` variable so I wouldn't be able to use it later in the `make-two-way-stream` call as the input stream; the pretty-printing stream doesn't support input, even if the original stream did, so I have to preserve the original

File: src/debugger.lisp (hunk 4 above)
Comment (goose121) [reply src/debugger.lisp]: Oops sorry I think that was my emacs and I didn't notice it

File: src/debugger.lisp (hunk 3 above)
Comment (phoe) [reply src/debugger.lisp:24]: TIL about `pprint-logical-block` rebinding the variable! Thank you.

**Comment by phoe** [comment]
Merging. Thanks for finding https://bugs.launchpad.net/sbcl/+bug/1909659 along the way!

**Review by phoe — APPROVED** [review]
